arrow/src/main/java/org/apache/iceberg/arrow/vectorized/NullabilityHolder.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.arrow.vectorized;
+
+public class NullabilityHolder {
+  private final byte[] isNull;
+  private int numNulls;
+
+  public NullabilityHolder(int size) {
+    this.isNull = new byte[size];
+  }
+
+  public void setNull(int idx) {
+    isNull[idx] = 1;
+    numNulls++;
```

Comment:
It doesn't look like `numNulls` is correct, since it is incremented here as well as in `setNotNull`. Even if `setNotNull` didn't increment it, the value would also be incorrect if the same position were set twice. Do we need to track the number of nulls? It doesn't look like this is called anywhere in this set of changes.